### PR TITLE
docs: add anchors for blogs

### PR DESCRIPTION
**Pull request by harish-sethuraman**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

Fixes https://github.com/eslint/eslint/issues/16128

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

Added anchor-js library and add anchors for headings inside docs

#### Is there anything you'd like reviewers to focus on?

check if any other element is getting anchors. I verified that table of content was h2 and was getting the anchor so I've move it to be a paragraph text and added font-weight alone to match the styling. If you are aware of any other elements like that need help. I verified once 😅  and everything seems to work fine

### CODE REVIEW

**Comment by nzakas** [comment]
Thanks for looking at this. Because this is only targeting blog posts, could all of the changes be made to layouts/post.html instead? That way everything that is post-specific is in one file.

**Comment by harish-sethuraman** [comment]
bump for review

File: src/assets/js/main.js [hunk 1]
```diff
@@ -89,3 +89,7 @@
         });
     }
 })();
+
+document.addEventListener("DOMContentLoaded", () => {
```
Comment (nzakas) [inline src/assets/js/main.js:93]: Can you also move this into post.html as an inline script? This causes an error on non-blog post pages now.

**Comment by nzakas** [comment]
@harish-sethuraman please don't bump for review. :) I already get notifications when pull requests are updated. If for some reason you don't get a response after a week, then it's appropriate to ask again.

File: src/assets/js/main.js (hunk 1 above)
Comment (harish-sethuraman) [reply src/assets/js/main.js]: Done

File: src/_includes/layouts/post.html [hunk 2]
```diff
@@ -8,7 +8,11 @@
 {%- from 'components/profile.macro.html' import member, contributor, post_author %}
 
 <script src="https://unpkg.com/anchor-js@4.3.1/anchor.min.js" type="text/javascript"></script>
-
+<script>
+    document.addEventListener("DOMContentLoaded", () => {
```
Comment (nzakas) [inline src/_includes/layouts/post.html:12]:
```suggestion
document.addEventListener("DOMContentLoaded", () => {
```

**Review by nzakas — APPROVED** [review]

**Comment by nzakas** [comment]
Thanks. Just a note: in this repo, we don’t prefix everything with “docs:”, that’s just in the eslint repo.

**Pull request merged**